www/assessment/items/itemsapi_assess.php

```diff
@@ -128,71 +150,362 @@
+         * @param  {object} items Object of all items currently loaded
+         * @return {object}       Current active item
+         */
+        function getActiveItem (items) {
+            for (var item in items) {
+                if (items.hasOwnProperty(item) && items[item].active === true) {
+                    return items[item];
+                }
+            }
+        }
+
+        function toggleModalClass () {
+            $('.modal-backdrop').css('display', 'none');
+        }
+        var topLeftRegion = [];
+        var topRightRegion = [];
+        var itemsRegion = [];
+        var rightRegion = [];
+        var bottomLeftRegion = [];
+        var bottomRightRegion = [];
+        var bottomRegion = [];
+        var region = {};
+        var regionElementMap = {
+            "pause_button": "Pause",
+            "accessibility_button": "Accessibility",
+            "calculator_button": "Calculator",
+            "flagitem_button": "Flag Item",
+            "fullscreen_button": "Full Screen",
+            "masking_button": "Response Masking",
+            "next_button": "Next",
+            "previous_button": "Pause",
+            "reviewscreen_button": "Review Screen",
+            "save_button": "Save",
+            "submit_button": "Submit",
+            "itemcount_element": "Item Count",
+            "reading_timer_element": "Reading Timer",
+            "timer_element": "Timer",
+            "title_element": "Title",
+            "progress_element": "Progress",
+            "slider_element": "Slider",
+            "vertical_element": "Vertical",
+            "verticaltoc_element": "Vertical TOC",
+            "horizontaltoc_element": "Horizontal TOC"
+
+        };
+        var regionElementToLabel = swap(regionElementMap);
+
+        function swap(json){
+            var ret = {};
+            for(var key in json){
+                ret[json[key]] = key;
+            }
+            return ret;
+        }
+        function removeElem(elem) {
+            $(elem).parent().parent().remove();
+
+        }
+
+        function addElem(selectElemID, container){
+            if($(selectElemID).val() != 'none'){
+                var text = $(selectElemID).val();
+                $(selectElemID).val("");
+                var elementLabelString = '<div class="elem-label"><span>' + regionElementMap[text] + '</span></div>';
+                var buttonString = '<div class="close-btn-container"><button class="close-btn" id="close-btn-'+btnIdCounter+ '">' + "×" + "</button></div>";
+                var divElem = '<div class="elem-container">' + elementLabelString + buttonString + "</div>";
+                $(container).append(divElem);
+                $('#close-btn-'+ btnIdCounter).click(function(){
+                    removeElem(this);
+                });
+                btnIdCounter++;
+                $(this).attr("disabled", "disabled");
+                $(this).removeAttr("disabled");
```

Comment:
This does not make sense to me as you add `disabled` then you remove it right away